### PR TITLE
[gcb] Add an interface to customize generateScopedName for CSSModule

### PR DESCRIPTION
**Pull request by callmedhh**

### CODE REVIEW

File: core-build/gulp-core-build-sass/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,10913 @@
+{
```
Comment (iclanton) [inline core-build/gulp-core-build-sass/package-lock.json:1]: This is a Rush repo. Delete this file.

You should install dependencies by running `rush install`.

File: common/changes/@microsoft/gulp-core-build-sass/yido-generateScopedName-interface_2020-07-01-06-56.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/gulp-core-build-sass",
+      "comment": "add an interface to allow customize generateScopedName for CSS modules.",
```
Comment (iclanton) [inline common/changes/@microsoft/gulp-core-build-sass/yido-generateScopedName-interface_2020-07-01-06-56.json:5]:
```suggestion
      "comment": "Allow customization of generated CSS module class names.",
```

File: core-build/gulp-core-build-sass/src/CSSModules.ts [hunk 3]
```diff
@@ -24,6 +24,7 @@ export interface ICSSModules {
 export default class CSSModules implements ICSSModules {
   private _classMap: IClassMap;
   private _rootPath: string;
+  private _generateScopedName: (name: string, fileName: string, css: string) => string;
```
Comment (iclanton) [inline core-build/gulp-core-build-sass/src/CSSModules.ts:27]:
```suggestion
  private _generateScopedName: ((name: string, fileName: string, css: string) => string) | undefined;
```

File: core-build/gulp-core-build-sass/src/CSSModules.ts [hunk 4]
```diff
@@ -33,19 +34,22 @@ export default class CSSModules implements ICSSModules {
...
+  public constructor(rootPath?: string, generateScopedName?: (name: string, fileName: string, css: string) => string) {
     this._classMap = {};
     if (rootPath) {
       this._rootPath = rootPath;
     } else {
       this._rootPath = process.cwd();
     }
+    if (generateScopedName) {
+      this._generateScopedName = generateScopedName;
+    }
```
Comment (iclanton) [inline core-build/gulp-core-build-sass/src/CSSModules.ts:46]:
```suggestion

    this._generateScopedName = generateScopedName;
```

File: core-build/gulp-core-build-sass/src/CSSModules.ts (hunk 3 above)
Comment (iclanton) [reply core-build/gulp-core-build-sass/src/CSSModules.ts:27]: It's a little awkward that this name is the same (save for the leading `_`) between the private and public members.

Comment (callmedhh) [reply core-build/gulp-core-build-sass/src/CSSModules.ts:27]: Good point, changed to `_customizedGenerateScopedName`

File: core-build/gulp-core-build-sass/package-lock.json (hunk 1 above)
Comment (callmedhh) [reply core-build/gulp-core-build-sass/package-lock.json]: Got it, thank you.

**Pull request merged**